feature/bgp/policybase/otg_tests/route_installation_test/route_installation_test.go

```diff
@@ -0,0 +1,629 @@
+	dstBgp6Peer.SetPeerAddress(dstIpv6.Gateway()).SetAsNumber(ateAS).SetAsType(gosnappi.BgpV6PeerAsType.EBGP)
+
+	dstBgp4PeerRoutes := dstBgp4Peer.V4Routes().Add().SetName(ateDst.Name + ".BGP4.peer" + ".RR4")
+	dstBgp4PeerRoutes.SetNextHopIpv4Address(dstIpv4.Address()).
+		SetNextHopAddressType(gosnappi.BgpV4RouteRangeNextHopAddressType.IPV4).
+		SetNextHopMode(gosnappi.BgpV4RouteRangeNextHopMode.MANUAL)
+	dstBgp4PeerRoutes.Addresses().Add().
+		SetAddress(advertisedRoutesv4Net).
+		SetPrefix(advertisedRoutesv4Prefix).
+		SetCount(routeCount)
+	dstBgp6PeerRoutes := dstBgp6Peer.V6Routes().Add().SetName(ateDst.Name + ".BGP6.peer" + ".rr6")
+	dstBgp6PeerRoutes.SetNextHopIpv6Address(dstIpv6.Address()).
+		SetNextHopAddressType(gosnappi.BgpV6RouteRangeNextHopAddressType.IPV6).
+		SetNextHopMode(gosnappi.BgpV6RouteRangeNextHopMode.MANUAL)
+	dstBgp6PeerRoutes.Addresses().Add().
+		SetAddress(advertisedRoutesv6Net).
+		SetPrefix(advertisedRoutesv6Prefix).
+		SetCount(routeCount)
+
+	// ATE Traffic Configuration
+	t.Logf("TestBGP:start ate Traffic config")
+	flowipv4 := config.Flows().Add().SetName("bgpv4RoutesFlow")
+	flowipv4.Metrics().SetEnable(true)
+	flowipv4.TxRx().Device().
+		SetTxNames([]string{srcIpv4.Name()}).
+		SetRxNames([]string{dstBgp4PeerRoutes.Name()})
+	flowipv4.Size().SetFixed(512)
+	flowipv4.Rate().SetPps(trafficPacketRate)
+	flowipv4.Duration().SetChoice("continuous")
+	e1 := flowipv4.Packet().Add().Ethernet()
+	e1.Src().SetValue(srcEth.Mac())
+	v4 := flowipv4.Packet().Add().Ipv4()
+	v4.Src().SetValue(srcIpv4.Address())
+	v4.Dst().Increment().SetStart(advertisedRoutesv4Net).SetCount(routeCount)
+
+	flowipv6 := config.Flows().Add().SetName("bgpv6RoutesFlow")
+	flowipv6.Metrics().SetEnable(true)
+	flowipv6.TxRx().Device().
+		SetTxNames([]string{srcIpv6.Name()}).
+		SetRxNames([]string{dstBgp6PeerRoutes.Name()})
+	flowipv6.Size().SetFixed(512)
+	flowipv6.Rate().SetPps(trafficPacketRate)
+	flowipv6.Duration().SetChoice("continuous")
+	e2 := flowipv6.Packet().Add().Ethernet()
+	e2.Src().SetValue(srcEth.Mac())
+	v6 := flowipv6.Packet().Add().Ipv6()
+	v6.Src().SetValue(srcIpv6.Address())
+	v6.Dst().Increment().SetStart(advertisedRoutesv6Net).SetCount(routeCount)
+
+	expected := otgutils.ExpectedState{
+		BGP4: map[string]otgutils.ExpectedBgpMetrics{
+			srcBgp4Peer.Name(): {Advertised: 0, Received: expectedRoutes},
+			dstBgp4Peer.Name(): {Advertised: routeCount, Received: 0},
+		},
+		BGP6: map[string]otgutils.ExpectedBgpMetrics{
+			srcBgp6Peer.Name(): {Advertised: 0, Received: expectedRoutes},
+			dstBgp6Peer.Name(): {Advertised: routeCount, Received: 0},
+		},
+	}
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	otg.PushConfig(t, config)
+	otg.StartProtocols(t)
+	return config, expected
+}
+
+// verifyTraffic confirms that every traffic flow has the expected amount of loss (0% or 100%
+// depending on wantLoss, +- 2%)
+func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, c gosnappi.Config, wantLoss bool) {
```

Comment:
Maybe there could be a flowutils package that checks the flow telemetry alone, that could be shared between ATE and OTG? otgutils.PrintFlowMetrics would be inlined to flowutils.VerifyTraffic().

The idea is that it just needs a list of flow names are arguments, and use (*ondatra.Device).Telemetry().Flow(flowName) to access the telemetry.

Reply:
Actually the util function only needs to check one flow at a time, and rather than asserting whether the traffic happened on behalf of the test, it should return whether the traffic passed as a bool.

```
package flowutils

func HasTraffic(t testing.TB, device *ondatra.Device, flowName string) bool {
  ...
}
```

We are moving away from the so-called assertion helpers. We can still have helpers that return the result of a check, but leave it up to the caller to decide what to do with that result.

Reply:
Okay, so if ATE and OTG telemetry are significantly different, then utils like HasTraffic can go into otgutils. But so far I've only seen that OTG is missing LossPct(), and haven't seen any other differences that could affect HasTraffic. I think for helpers that aren't affected by the ATE and OTG telemetry differences, it would probably be best to make them work for both and avoid ATE-ism or OTG-ism.

I still think the helper should avoid having the for loop over all flows. The helper should check one flow at a time, and the caller will call it using explicit flow names.

Reply:
According to the latest reviews on PR#259 now we have only print utilities under otgutils package. In that context may we request your approval on this PR#240.